src/http/client.rs

```diff
@@ -2828,12 +2828,12 @@ impl Http {
     pub async fn get_channel_archived_public_threads(
         &self,
         channel_id: ChannelId,
-        before: Option<u64>,
+        before: Option<Timestamp>,
         limit: Option<u64>,
     ) -> Result<ThreadsData> {
         let mut params = vec![];
         if let Some(before) = before {
-            params.push(("before", before.to_string()));
+            params.push(("before", before.to_rfc3339().unwrap()));
```

Comment:
The implementation of `Display` for `Timestamp` already calls `to_rfc3339`, so we don't actually need to change anything here.